sdk/azcore/internal/exported/request.go

```diff
@@ -102,24 +102,28 @@ func (req *Request) OperationValue(value interface{}) bool {
 
 // SetBody sets the specified ReadSeekCloser as the HTTP request body.
```

Comment:
If we decide to keep the functionality in `SetBody()` then we should document this part of the behavior.

Reply:
Well, it got a little more interesting. RFC 9110 specifies Content-Length shouldn't be set when a request contains no content, so I now have `SetBody()` remove the header in that case. The RFC has no similar guidance for Content-Type. I see that Go will send a 0-byte value for Content-Type when the header is set to "". That's legal HTTP and shouldn't make a difference to the server, but it means the current API can express either "I want to unset Content-Type" or "I want to set Content-Type to `""`". I prefer the latter--`SetBody()` should continue setting whatever value the caller provides and not have an opinion about what that value means. If it turns out someone really needs to unset Content-Type, they can work around this by modifying the underlying `http.Request`.

Reply:
Good catch on the Content-Length.

Regarding Content-Type, I get what you're saying and mostly agree with you.  I just think it's a bit strange that if we're saying "pass nil/empty body to remove the body" that it doesn't also remove the Content-Type header but sends a zero-value instead.  I get that in practice it shouldn't be a problem, so maybe I'm overthinking it.

Reply:
I've definitely overthought it, and you have me reconsidering again, so I'll leave this PR open until tomorrow. At this moment I don't mind sending an empty Content-Type because although that wastes a few bytes, it is valid and semantically correct, and I believe the resulting behavior is easier to reason about.